### DISCUSSION TITLE
Allow setting device info directly in Events API without user agent

### DISCUSSION THREAD

**Discussion by hig-dev** [plausible/analytics#2726] [Anything Else]
Hello,

I am trying to integrate Plausible Analytics into my Flutter mobile app.
I am using the Events API (https://plausible.io/docs/events-api) to send `pageview` events, however, the OS and screen size information is "not set" on my dashboard.

It would be really helpful if you could add an optional request body JSON parameter for custom device information.

Without this feature, it is necessary to somehow create a user agent that gets correctly parsed but I did not find an easy way to do that reliabily.

**Comment by metmarkosaric** [comment]
thanks for the feedback @hig-dev! i do know that there's a Flutter plugin, i wonder if it helps you? https://plausible.io/docs/integration-guides#flutter

**Reply by hig-dev** [reply]
Yes, I know and I am using it: https://pub.dev/packages/plausible_analytics

However: the plugin generates an user agent that does not work with the device parsing of Plausible

For developers who want to use Plausible in mobile apps it does not make sense to generate an user agent.

Currently:
1. The app developer must generate an user agent that works with the device parsing by Plausible
2. Plausible has to parse the user agent to get device details

My proposition:
1. The app developer has the option to set custom device information in the request (e.g. in the props parameter)
2. Plausible does not have to parse the user agent if the custom device information is set
